shop.go

```diff
@@ -39,20 +58,23 @@ func NewShop(shop string, accessToken string) Shop {
 	This constructor automatically uses the URL passed to it to communicate with the store.
 	If you would like to use an auto store URL use the `NewShop` constructor instread.
 	Example:
-	shop := shopify.NewCustomShop("https://my-shop-domain.com/foo/bar", "shppy_21u92h2184ho912h29r01")
+	shop := shopify.NewCustomShop("https://my-shop-domain.com/foo/bar", "shppy_21u92h2184ho912h29r01", httpshopify.IsPlus)
 	shippedOrders, err := shop.Orders().List(shopify.OrderQuery{FulfillmentStatus:"shipped"})
 	For the full shopify admin REST API documentation see https://shopify.dev/docs/admin-api/rest/reference
 */
-func NewCustomShop(url string, accessToken string) Shop {
-	client := http.NewClient(http.RequestHeaders{
-		{
-			Name:  "X-Shopify-Access-Token",
-			Value: accessToken,
-		}, {
-			Name:  "Content-Type",
-			Value: "application/json",
-		},
-	})
+func NewCustomShop(url string, accessToken string, isPlus bool) Shop {
+	var rateLimitOption http.Option
```

Comment:
Could also be written:

```
	rateLimitOption := RateLimitDefault()
	
	if isPlus {
		rateLimitOption = RateLimitPlus()
	}
```

Although the approach for the intial commit is more idomatic

Reply:
I'm happy either way, which would you prefer?

Reply:
Keep the first as it is more idiomatic, I would write the second but it is not very go like.